sktime/forecasting/base/_base.py

```diff
@@ -592,14 +502,69 @@ def _set_y_X(self, y, X=None, enforce_index_type=None):
         # set initial cutoff to the end of the training data
         self._set_cutoff_from_y(y)
 
+    def _check_X_y(self, y=None, X=None):
+        """Check and coerce X/y for fit/predict/update functions."""
+        # input checks and minor coercions on X, y
+        ###########################################
+
+        enforce_univariate = self.get_tag("scitype:y") == "univariate"
+        enforce_multivariate = self.get_tag("scitype:y") == "multivariate"
+        enforce_index_type = self.get_tag("enforce_index_type")
+
+        # checking y
+        if y is not None:
+            check_y_args = {
+                "enforce_univariate": enforce_univariate,
+                "enforce_multivariate": enforce_multivariate,
+                "enforce_index_type": enforce_index_type,
+                "allow_None": False,
+            }
+
+            y = check_series(y, **check_y_args, var_name="y")
+
+            self._y_mtype_last_seen = mtype(y)
+        # end checking y
+
+        # checking X
+        if X is not None:
+            X = check_series(X, enforce_index_type=enforce_index_type, var_name="X")
+            if self.get_tag("X-y-must-have-same-index"):
+                check_equal_time_index(X, y)
+        # end checking X
+
+        # convert y to supported inner type, if necessary
+        ##################################################
+
+        # retrieve supported mtypes
+
+        # convert X and y to a supported internal mtype
+        #  it X/y mtype is already supported, no conversion takes place
+        #  if X/y is None, then no conversion takes place (returns None)
+        y_inner_mtype = self.get_tag("y_inner_mtype")
+        y_inner = convert_to(
+            y,
+            to_type=y_inner_mtype,
+            as_scitype="Series",  # we are dealing with series
+            store=self.converter_store_y,
```

Comment:
As discussed, let's make this private
```suggestion
            store=self._converter_store_y,
```

Reply:
done